emr_cost_calculator.py

```diff
@@ -48,89 +43,131 @@ def validate_date(date_text):
+    def __init__(self, group_id, instance_type, market_type, group_type):
         self.group_id = group_id
         self.instance_type = instance_type
+        self.market_type = market_type
         self.group_type = group_type
-        self.price = 0
 
 
-class EmrCostCalculator:
+class Ec2EmrPricing:
+    def __init__(self, region):
+        url_base = 'https://pricing.us-east-1.amazonaws.com'
+
+        index_response = requests.get(url_base + '/offers/v1.0/aws/index.json')
+        index = index_response.json()
+
+        emr_regions_response = requests.get(url_base + index['offers']['ElasticMapReduce']['currentRegionIndexUrl'])
+        emr_region_url = url_base + emr_regions_response.json()['regions'][region]['currentVersionUrl']
+
+        emr_pricing = requests.get(emr_region_url).json()
+        sku_to_instance_type = {}
+        for sku in emr_pricing['products']:
+            if emr_pricing['products'][sku]['attributes']['softwareType'] == 'EMR':
+                sku_to_instance_type[sku] = emr_pricing['products'][sku]['attributes']['instanceType']
+
+        self.emr_prices = {}
+        for sku in sku_to_instance_type.keys():
+            instance_type = sku_to_instance_type.get(sku)
+            price = float(emr_pricing['terms']['OnDemand'][sku].itervalues().next()['priceDimensions'].itervalues().next()['pricePerUnit']['USD'])
+            self.emr_prices[instance_type] = price
 
+        ec2_regions_response = requests.get(url_base + index['offers']['AmazonEC2']['currentRegionIndexUrl'])
+        ec2_region_url = url_base + ec2_regions_response.json()['regions'][region]['currentVersionUrl']
+
+        ec2_pricing = requests.get(ec2_region_url).json()
+
+        ec2_sku_to_instance_type = {}
+        for sku in ec2_pricing['products']:
+            try:
+                if ec2_pricing['products'][sku]['attributes']['tenancy'] == 'Shared' \
+                        and ec2_pricing['products'][sku]['attributes']['operatingSystem'] == 'Linux':
+                    ec2_sku_to_instance_type[sku] = ec2_pricing['products'][sku]['attributes']['instanceType']
+
+            except KeyError:
+                pass
+
+        self.ec2_prices = {}
+        for sku in ec2_sku_to_instance_type.keys():
+            instance_type = ec2_sku_to_instance_type.get(sku)
+            price = float(ec2_pricing['terms']['OnDemand'][sku].itervalues().next()['priceDimensions'].itervalues().next()['pricePerUnit']['USD'])
+            self.ec2_prices[instance_type] = price
+
+    def get_emr_price(self, instance_type):
+        return self.emr_prices[instance_type]
+
+    def get_ec2_price(self, instance_type):
+        return self.ec2_prices[instance_type]
+
+
+class EmrCostCalculator:
     def __init__(
-        self,
-        region,
-        aws_access_key_id=None,
-        aws_secret_access_key=None
-    ):
+            self,
+            region,
+            aws_access_key_id=None,
+            aws_secret_access_key=None):
```

Comment:
can you please keep the indentation consistent with the rest of the file aka:
```python
def __init__(
        self,
        region,
        aws_access_key_id=None,
        aws_secret_access_key=None
    ):
```
notice ending parentesis is in new line.


*similarly for the rest of the file.*

Reply:
Also I think you have intended one level too far.